### PR TITLE
Add """ ... """ as a long string format in VCL

### PR DESCRIPTION
**Pull request by andrewwiik**
This PR adds """ ... """ as a long string format to VCL. The purpose of this is to allow JSON to be more easily written in VCL as the current string format(s) in VCL heavily conflict with JSON. While I wanted to use backticks (`) @bsdphk said we should go with triple double quotes to keep in line with borrowing things from python.

### CODE REVIEW

File: bin/varnishtest/tests/v00019.vtc [hunk 1]
```diff
@@ -18,6 +18,11 @@ varnish v1 -errvcl {Unterminated long-string, starting at} {
 	{" }
 }
 
+varnish v1 -errvcl {Unterminated long-string, starting at} {
+	backend b { .host = "127.0.0.1"; }
+	""" ""
```
Comment (gquintard) [inline bin/varnishtest/tests/v00019.vtc:23]: isn't the problem here that even if the string was fine, it'd be in the wrong place. In other words: are you testing the right thing?

File: bin/varnishtest/tests/v00066.vtc [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+varnish v1 -vcl {
+	backend default none;
+
+	sub vcl_recv {
+		return (synth(200));
+	}
+
+	sub vcl_synth {
+                set resp.body = """{"key":"value"}""";
```
Comment (gquintard) [inline bin/varnishtest/tests/v00066.vtc:11]: this needs more `\n` and `{"`

File: bin/varnishtest/tests/v00019.vtc (hunk 1 above)
Comment (andrewwiik) [reply bin/varnishtest/tests/v00019.vtc:23]: According to the previous test no as we are testing throwing this specific error.

File: bin/varnishtest/tests/v00066.vtc (hunk 2 above)
Comment (andrewwiik) [reply bin/varnishtest/tests/v00066.vtc:11]: This is already tested in the built-in VCL for {", if you think a different string for the test would be better please let me know what that string would be.

Comment (gquintard) [reply bin/varnishtest/tests/v00066.vtc:11]:
```
std.log("""
{" "} "  
""");
```

would work for me, but `varnishtest` may not cooperate.

File: bin/varnishtest/tests/v00019.vtc (hunk 1 above)
Comment (gquintard) [reply bin/varnishtest/tests/v00019.vtc:23]: fair enough, I missed that you are checking the error string too

File: bin/varnishtest/tests/v00066.vtc (hunk 2 above)
Comment (Dridi) [reply bin/varnishtest/tests/v00066.vtc:11]: If you have a `{...}` balance issue in `varnishtest` multi-line strings you can often work around it with VCL:

```
varnish v1 -vcl+backend {
    import std;
    sub vcl_recv {
        std.log("{");
        # adding '}' for balance
    }
}
```

Add the right brace at the right location in a comment.

**Review by Dridi — APPROVED** [review]

**Comment by Dridi** [comment]
I haven't decided whether it would be worth it to fold `{"..."}` and `"""..."""` parsing together since it's the exact same code.

**Comment by andrewwiik** [comment]
@Dridi I thought about that too but there are a lot of places where you need to either add multiples of 2 or 3 and I thought doing a lot of ternary operators would be annoying there.

**Comment by nigoroll** [comment]
bugwash: add docs, then merge

**Comment by andrewwiik** [comment]
@Dridi merge me pls :)

**Comment by Dridi** [comment]
7a12b1c3ff...3ec74a3aa2

**Pull request closed without merging**